src/server/client.rs

```diff
@@ -178,13 +179,37 @@ async fn handle_worker_stop(
     ToClientMessage::StopWorkerResponse(responses)
 }
 
-fn compute_job_detail(state_ref: &StateRef, job_id: JobId, include_tasks: bool) -> ToClientMessage {
+fn compute_job_detail(
+    state_ref: &StateRef,
+    selector: Selector,
+    include_tasks: bool,
+) -> ToClientMessage {
     let state = state_ref.get();
-    ToClientMessage::JobDetailResponse(
-        state
+
+    let job_ids: Vec<JobId> = match selector {
+        Selector::All => state_ref
+            .get()
+            .jobs()
+            .map(|job| job.make_job_info())
+            .map(|job_info| job_info.id)
```

Comment:
These to maps could by written just as ``.map(|job| job.job_id)`` without creating job info